src/read.c

```diff
@@ -5304,25 +5290,26 @@ static uint32_t avifGetDecodedRowCount(const avifDecoder * decoder, const avifTi
 
 uint32_t avifDecoderDecodedRowCount(const avifDecoder * decoder)
 {
-    const uint32_t colorRowCount = avifGetDecodedRowCount(decoder, &decoder->data->color, decoder->image);
-    if (colorRowCount == 0) {
-        return 0;
-    }
-    const uint32_t alphaRowCount = avifGetDecodedRowCount(decoder, &decoder->data->alpha, decoder->image);
-    if (alphaRowCount == 0) {
-        return 0;
+    uint32_t minRowCount = decoder->image->height;
+    for (int c = 0; c < AVIF_ITEM_CATEGORY_COUNT; ++c) {
```

Comment:
Shouldn't you skip the gain map in this loop? since there is special logic for it aferwards. It seems the count will be underestimated if the gainmap is smaller than the mai image.